hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -99,6 +100,8 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
 
   protected val sparkSession: SparkSession = sqlContext.sparkSession
 
+  protected lazy val tableName = metaClient.getTableConfig.getTableName
```

Comment:
nit: better make this a method so that we don't carry this field additional when serializing

Reply:
fixed